django/core/checks/urls.py

```diff
@@ -19,23 +19,15 @@ def check_resolver(resolver):
     """
     Recursively check the resolver.
     """
-    from django.urls import RegexURLPattern, RegexURLResolver
-    warnings = []
-    for pattern in resolver.url_patterns:
-        if isinstance(pattern, RegexURLResolver):
-            warnings.extend(check_include_trailing_dollar(pattern))
-            # Check resolver recursively
-            warnings.extend(check_resolver(pattern))
-        elif isinstance(pattern, RegexURLPattern):
-            warnings.extend(check_pattern_name(pattern))
-        else:
-            # This is not a url() instance
-            warnings.extend(get_warning_for_invalid_pattern(pattern))
-
-        if not warnings:
-            warnings.extend(check_pattern_startswith_slash(pattern))
-
-    return warnings
+    check_method = getattr(resolver, "check", None)
+    resolve_method = getattr(resolver, "resolve", None)
+    reverse_method = getattr(resolver, "reverse", None)
+    if check_method is not None:
+        return check_method()
+    elif resolve_method is None and reverse_method is None:
```

Comment:
I think you should either make that an `or`, or drop checking `reverse_method`.

A resolver without `resolve` makes no sense.
A resolver without `reverse` might.